chart/templates/configmap.yaml

```diff
@@ -0,0 +1,108 @@
+################################
+## Airflow ConfigMap
+#################################
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "airflow_config" . }}
+  labels:
+    tier: airflow
+    component: config
+    release: {{ .Release.Name }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    heritage: {{ .Release.Service }}
+{{- with .Values.labels }}
+{{ toYaml . | indent 4 }}
+{{- end }}
+data:
+  # These are system-specified config overrides.
+  airflow.cfg: |
```

Comment:
can we expose the airflow.cfg via helm values 

```
airflow_cfg:
  core:
    dags_folder: '$AIRFLOW_HOME/dags/repo'
    base_log_folder: '$AIRFLOW_HOME/logs'
    executor: KubernetesExecutor
    plugins_folder: '$AIRFLOW_HOME/plugins'
    sql_alchemy_conn:  $sql_alchemy_conn
    load_examples: false
    remote_logging:  false
    logging_level: WARN
    fab_logging_level: WARN
    store_serialized_dags: True
    min_serialized_dag_update_interval: 30
  scheduler:
    child_process_log_directory:  '$AIRFLOW_HOME/logs/scheduler'
    catchup_by_default: False
  webserver:
    rbac:  True
    update_fab_perms: True
    enable_proxy_fix: True
```


and render it using something like the below one
```
  airflow.cfg: |-
    {{- $root := . -}}
    {{- range $airflow_config_header, $config_dict := .Values.airflow_cfg }}
    [{{ $airflow_config_header }}]
    {{- range $key, $val := $config_dict }}
    {{ $key }} = {{ $val }}
    {{- end}}
    {{- end}}
```

to make it easier to change the airflow configuration

Reply:
Perhaps, but you can set all config options by environment variables: `AIRFLOW__CORE__STORE_SERIALIZED_DAGS` (for instance) via the `env` top level key in Values  so it's not needed. Might be a good idea, but not needed

Reply:
I envisioned this default `airflow.cfg` as a good starting point for this chart and that any user-specific overrides would be passed via the `env` field as @ashb mentioned. Mostly so that we don't need to maintain a helm value to airflow.cfg mapping.

Reply:
@schnie @ashb  a follow up question on this one. `.Values.env` will work for scheduler, webserver and workers when we use the `CeleryExecutor` , but how is it handled when user opts for   `KubernetesExecutor` ? the configmap doesn't seem to be update `[kubernetes_environment_variables]` of `airflow.cfg` with  the the custom environment variables set by the user